lib/_http_incoming.js

```diff
@@ -67,6 +69,8 @@ IncomingMessage.prototype.setTimeout = function(msecs, callback) {
 
 
 IncomingMessage.prototype.read = function(n) {
+  if (!this._consuming)
+    this._readableState.readingMore = false;
```

Comment:
Yeah, I believe it would make a difference. `_consuming ` can flip only into `true`, and can't really go back. This is a hack to detect following scenario:

```js
http.createServer((req, res) => {
  res.end('123');
});
```

Clearly in this case no one has attempted to read from `req`. Thus `_consuming` is `false` here, and in `_http_server.js` the request can be `_dump`ed. However, initial version (prior to this patch) of this hack wasn't working as expected, because `_stream_readable.js` may call `.read()` by itself. Luckily it calls it only from `maybeReadMore`, so we could test it here.